rpc_jobs/unit/jira.yml

```diff
@@ -0,0 +1,132 @@
+- job:
+    name: RE-unit-test-jira
+    project-type: pipeline
+    concurrent: true
+    properties:
+      - build-discarder:
+          num-to-keep: 30
+    parameters:
+      - rpc_gating_params
+    dsl: |
+      def cleanup(String issue, String issue_query){
+        // remove build specific label to avoid adding an extra label
+        // to jira every time this job runs.
+        common.jira_set_labels(issue, base_labels)
+        common.jira_close(issue)
+        issues = common.jira_query(issue_query)
+        print issues
+        assert issues.size() == 0
+      }
+
+      library "rpc-gating@${RPC_GATING_BRANCH}"
+      common.shared_slave{
+        project = "SAN" // sandbox project
+        base_labels = ['re_unit', 'RE-unit-test-jira', 'jenkins']
+        labels = base_labels + env.BUILD_TAG
+        issue_query="labels = \"re_unit\" AND labels = \"${env.BUILD_TAG}\" and project = ${project} and status = BACKLOG"
+
+        try {
+          stage("Test query command with issue key"){
+            String key = "RE-3"
```

Comment:
I'm not sure we should be using issues from real projects in the tests. I'd rather this was done using the sandbox project, maybe even create one per run to save having a hardcoded issue.

Reply:
I'd prefer to use a hard coded issue so that the query function can be tested independently of the create function. The reason being that the query function is then used to verify the various create functions. Changed to SAN-1.